inttests/volume_test.go

```diff
@@ -476,3 +548,74 @@ func TestSetMappedSdcLimitsInvalid(t *testing.T) {
 	assert.Nil(t, err)
 	deleteAllVolumes(t)
 }
+
+// Testing TestLockUnlockAutoSnapshot will attempting locking the auto snapshot and unlocking the auto snapshot
+func TestLockUnlockAutoSnapshot(t *testing.T) {
+	volID, err := createVolume(t, "")
+	assert.Nil(t, err)
+	newVolume, err := getVolByID(volID)
+	assert.Nil(t, err)
+
+	// now make a snapshot
+	snapshotDefs := make([]*siotypes.SnapshotDef, 0)
+	snapname := fmt.Sprintf("%s-%s", newVolume.Name, "snap")
+
+	snapDef := &siotypes.SnapshotDef{
+		VolumeID:     volID,
+		SnapshotName: snapname,
+	}
+	snapshotDefs = append(snapshotDefs, snapDef)
+	snapParam := &siotypes.SnapshotVolumesParam{
+		SnapshotDefs: snapshotDefs,
+	}
+
+	system := getSystem()
+	assert.NotNil(t, system)
+
+	// Create snapshot
+	snapResponse, err := system.CreateSnapshotConsistencyGroup(snapParam)
+	assert.Nil(t, err)
+	assert.NotZero(t, len(snapResponse.VolumeIDList))
+
+	// Get StoragePool
+	pool := getStoragePool(t)
+	volumes, err := pool.GetVolume("", volID, "", "", true)
+	assert.Nil(t, err)
+	assert.NotNil(t, volumes)
+
+	// Get Snapshot
+	volumes, err = pool.GetVolume("", snapResponse.VolumeIDList[0], "", "", true)
+	assert.Nil(t, err)
+	assert.NotNil(t, volumes)
+
+	// Lock Snapshot
+	snap, err := getVolByID(volumes[0].ID)
+	assert.Nil(t, err)
+	sr := goscaleio.NewVolume(C)
+	sr.Volume = snap
+	err = sr.LockAutoSnapshot()
+	assert.NotNil(t, err)
+	err = sr.UnlockAutoSnapshot()
+	assert.NotNil(t, err)
+
+	// Delete Snapshot and Volume
+	err = deleteVolume(t, sr.Volume.ID)
+	assert.Nil(t, err)
+	err = deleteVolume(t, volID)
+	assert.Nil(t, err)
+	deleteAllVolumes(t)
+}
+
+// Testing TestSetVolumeAccessModeLimit will be attempting set access mode of volume
+func TestSetVolumeAccessModeLimit(t *testing.T) {
+	volID, err := createVolume(t, "")
+	assert.Nil(t, err)
+	vol, err := getVolByID(volID)
+	assert.Nil(t, err)
+	vr := goscaleio.NewVolume(C)
+	vr.Volume = vol
+	err = vr.SetVolumeAccessModeLimit("ReadOnly")
```

Comment:
Can you please cover some negative test scenarios as well.

Reply:
![image](https://user-images.githubusercontent.com/118425422/207311037-e30baadd-f41e-4d8e-8b69-9b4b4767c040.png)
added the negative test for the above scenario